packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationUtilities.m

```diff
@@ -0,0 +1,287 @@
+{
+  NSArray<NSString *> *devicesWithNotchIdentifiers = @[
+                                                       @"iPhone10,3", // iPhoneX
+                                                       @"iPhone10,6", // iPhoneX
+                                                       @"iPhone11,2", // iPhoneXs
+                                                       @"iPhone11,6", // iPhoneXsMax
+                                                       @"iPhone11,4", // iPhoneXsMax
+                                                       @"iPhone11,8", // iPhoneXr
+                                                       ];
+  NSArray<NSString *> *simulatorsIdentifiers = @[
+                                                 @"i386",
+                                                 @"x86_64",
+                                                 ];
+  
+  if ([devicesWithNotchIdentifiers containsObject:deviceIdentifier]) {
+    return YES;
+  }
+  
+  if ([simulatorsIdentifiers containsObject:deviceIdentifier]) {
+    return [self doesDeviceHaveNotch:[[[NSProcessInfo processInfo] environment] objectForKey:@"SIMULATOR_MODEL_IDENTIFIER"]];
+  }
+  return NO;
+}
+
++ (NSDictionary *)getStringToOrientationJSDict
+{
+  static NSDictionary*strToOrientationDict = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    strToOrientationDict = @{@"PORTRAIT" : @(EXOrientationPortrait),
+                             @"PORTRAIT_UP" : @(EXOrientationPortraitUp),
+                             @"PORTRAIT_DOWN" : @(EXOrientationPortraitDown),
+                             @"LANDSCAPE" : @(EXOrientationLandscape),
+                             @"LANDSCAPE_LEFT" : @(EXOrientationLandscapeLeft),
+                             @"LANDSCAPE_RIGHT" : @(EXOrientationLandscapeRight),
+                             @"UNKNOWN": @(EXOrientationUnknown)
+                             };
+  });
+  return strToOrientationDict;
+}
+
++ (NSDictionary *)getOrientationJSToStringDict
+{
+  static NSMutableDictionary*orientationToStrDict = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationToStrDict = [[NSMutableDictionary alloc] init];
+    NSDictionary *strToOrientation = [EXScreenOrientationUtilities getStringToOrientationJSDict];
+    for(NSString *str in strToOrientation) {
+      NSNumber *wrappedOrientation = [strToOrientation objectForKey:str];
+      orientationToStrDict[wrappedOrientation] = str;
+    }
+  });
+  return orientationToStrDict;
+}
+
++ (NSDictionary *)getStringToOrientationLockJSDict
+{
+  static NSDictionary*strToOrientationLockDict = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    strToOrientationLockDict = @{ @"DEFAULT" : @(EXOrientationDefaultLock),
+                                  @"ALL" : @(EXOrientationAllLock),
+                                  @"PORTRAIT" : @(EXOrientationPortraitLock),
+                                  @"PORTRAIT_UP" : @(EXOrientationPortraitUpLock),
```

Comment:
Do you think it would be possible to get rid of `EXOrientation` concept in favor of checking with `UIInterface` (and `EXOrientationLock` in favor of `UIInterfaceLock`)?

Reply:
My question comes from the feeling that there's just too much converting in these classes in different places (notice there are for orientation: `uideviceorientation`, `uiinterfaceorientation`, `exorientation`, `exorientation in string form`) I know this is a very generic comment, but maybe you have some ideas on how to simplify the logic?